### PR TITLE
Icon preview / upload

### PR DESCRIPTION
**Pull request by RickCarlino**
This PR adds the ability to add an Icon to a crop from the edit page.

Also, there were some formatting fixes that my editor added in (only for the files I was working on).

### CODE REVIEW

File: app/assets/javascripts/crops/svg_button.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+var MISSING_PROP =
+  "You forgot to attach an `on-change=''` function to the <svg-button/> directive.";
```
Comment (simonv3) [inline app/assets/javascripts/crops/svg_button.js]: Huh, I like this.

File: app/assets/javascripts/crops/svg_button.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      var $ctrl = this;
+      var btn = $element.find("input")[0];
+      btn
+        .addEventListener("change", function (e) {
+          var file = e.currentTarget.files[0];
+          var reader = new FileReader();
+          reader.onload = function (e) {
+            var result = { $svg: reader.result };
+            $ctrl.onChange ? $ctrl.onChange(result)
+              : console.error(MISSING_PROP);
```
Comment (simonv3) [inline app/assets/javascripts/crops/svg_button.js]: Yeah for readability purposes can you do this in an `if else`?

Comment (RickCarlino) [inline app/assets/javascripts/crops/svg_button.js]: Sure thing

File: app/assets/javascripts/crops/svg_button.js (hunk 1 above)
Comment (RickCarlino) [inline app/assets/javascripts/crops/svg_button.js]: Yeah, I'm always thrown off by silent errors with missing props.

File: app/assets/javascripts/crops/edit.js [hunk 3]
```diff
@@ -69,16 +70,20 @@ openFarmApp.controller('cropCtrl', ['$scope', '$http', 'cropService',
...
+          crop,
+          cropCallback, function (err) {
+            console.log('err', err);
+          });
       }
     };
+    $scope.addSvg = function ($svg) {
+      $scope.crop.svg_icon = $svg;
+      console.dir($scope.crop);
+    }
```
Comment (RickCarlino) [inline app/assets/javascripts/crops/edit.js]: :thinking: I don't think so, according to [this validator](http://esprima.org/demo/validate.html)???

Comment (Br3nda) [inline app/assets/javascripts/crops/edit.js]: you've got the missing ) above in the same file (line 8), which causes a syntax error to land here.

**Pull request merged**